gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/strategy/optimization/RepeatUnrollStrategy.java

```diff
@@ -56,8 +62,14 @@
 public final class RepeatUnrollStrategy extends AbstractTraversalStrategy<TraversalStrategy.OptimizationStrategy> implements TraversalStrategy.OptimizationStrategy {
 
     private static final RepeatUnrollStrategy INSTANCE = new RepeatUnrollStrategy();
-    protected static final int MAX_BARRIER_SIZE = 2500;
-    private static final Set<Class> INVALIDATING_STEPS = new HashSet<>(Arrays.asList(LambdaHolder.class, LoopsStep.class));
+    static final int MAX_BARRIER_SIZE = 2500;
+    private static final Set<Class> ALLOWED_STEP_CLASSES = Set.of(
+            VertexStepContract.class,
+            EdgeVertexStep.class,
+            EdgeOtherVertexStep.class,
+            HasStep.class,
+            RepeatStep.class,
```

Comment:
Does this imply that the strategy will unroll nested repeats? Is this intended and if so, could you add test cases for this?